pkg/audit/cursor.go

```diff
@@ -128,11 +110,43 @@ func getRandomStripe(pointer *pb.Pointer) (index int64, err error) {
 	return randomStripeIndex.Int64(), nil
 }
 
-func getRandomPointer(pointerItems []*pb.ListResponse_Item) (pointer *pb.ListResponse_Item, err error) {
+// getRandomPointer attempts to get a random pointer from a list, and if the selected pointer is invalid, it returns a list without that item
+func (cursor *Cursor) getRandomPointer(pointerItems []*pb.ListResponse_Item) (pointer *pb.Pointer, newPointerItems []*pb.ListResponse_Item, err error) {
 	randomNum, err := rand.Int(rand.Reader, big.NewInt(int64(len(pointerItems))))
 	if err != nil {
-		return &pb.ListResponse_Item{}, err
+		return nil, pointerItems, err
+	}
+
+	i := randomNum.Int64()
+	pointerItem := pointerItems[i]
+	path := pointerItem.Path
+
+	// get pointer info
+	pointer, err = cursor.pointerdb.Get(path)
+	if err != nil {
+		return nil, pointerItems, err
+	}
+
+	newPointerItems = append(pointerItems[:i], pointerItems[i+1:]...)
+
+	//delete expired items rather than auditing them
+	if expiration := pointer.GetExpirationDate(); expiration != nil {
+		t, err := ptypes.Timestamp(expiration)
+		if err != nil {
+			return nil, newPointerItems, err
+		}
+		if t.Before(time.Now()) {
+			return nil, newPointerItems, cursor.pointerdb.Delete(path)
+		}
+	}
+
+	if pointer.GetType() != pb.Pointer_REMOTE {
+		return nil, newPointerItems, Error.New("not remote pointer")
+	}
+
+	if pointer.GetSegmentSize() == 0 {
+		return nil, newPointerItems, Error.New("segment size 0")
```

Comment:
I feel like the error messages for these cases might be more useful as logs instead of compiled into the error group and returned when no pointers are left, but also I'm kind of inventing a situation where they would be necessary as logs. Seems fine!

Reply:
I got rid of errors for these cases since they are not really "errors"